In [1]:
# importing the requried librabies
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
# creating a dataframe 
train_df=pd.read_csv(r"C:\Users\admin\Downloads\train_df.csv")
test_df=pd.read_csv(r"C:\Users\admin\Downloads\test_df.csv")

C:\Users\admin\AppData\Local\Temp\ipykernel_4056\3414869663.py:2: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.
  train_df=pd.read_csv(r"C:\Users\admin\Downloads\train_df.csv")


In [3]:
#printing top 10 rows of the dataframe
train_df.head(10)


Id County Province_State Country_Region  Population    Weight        Date  \
0   1    NaN            NaN    Afghanistan    27657145  0.058359  23-01-2020   
1   2    NaN            NaN    Afghanistan    27657145  0.583587  23-01-2020   
2   3    NaN            NaN    Afghanistan    27657145  0.058359  24-01-2020   
3   4    NaN            NaN    Afghanistan    27657145  0.583587  24-01-2020   
4   5    NaN            NaN    Afghanistan    27657145  0.058359  25-01-2020   
5   6    NaN            NaN    Afghanistan    27657145  0.583587  25-01-2020   
6   7    NaN            NaN    Afghanistan    27657145  0.058359  26-01-2020   
7   8    NaN            NaN    Afghanistan    27657145  0.583587  26-01-2020   
8   9    NaN            NaN    Afghanistan    27657145  0.058359  27-01-2020   
9  10    NaN            NaN    Afghanistan    27657145  0.583587  27-01-2020   

           Target  TargetValue  
0  ConfirmedCases            0  
1      Fatalities            0  
2  ConfirmedCases            0  
3      Fatalities            0  
4  ConfirmedCases            0  
5      Fatalities            0  
6  ConfirmedCases            0  
7      Fatalities            0  
8  ConfirmedCases            0  
9      Fatalities            0

In [4]:
test_df.head(10)

ForecastId County Province_State Country_Region  Population    Weight  \
0           1    NaN            NaN    Afghanistan    27657145  0.058359   
1           2    NaN            NaN    Afghanistan    27657145  0.583587   
2           3    NaN            NaN    Afghanistan    27657145  0.058359   
3           4    NaN            NaN    Afghanistan    27657145  0.583587   
4           5    NaN            NaN    Afghanistan    27657145  0.058359   
5           6    NaN            NaN    Afghanistan    27657145  0.583587   
6           7    NaN            NaN    Afghanistan    27657145  0.058359   
7           8    NaN            NaN    Afghanistan    27657145  0.583587   
8           9    NaN            NaN    Afghanistan    27657145  0.058359   
9          10    NaN            NaN    Afghanistan    27657145  0.583587   

         Date          Target  
0  27-04-2020  ConfirmedCases  
1  27-04-2020      Fatalities  
2  28-04-2020  ConfirmedCases  
3  28-04-2020      Fatalities  
4  29-04-2020  ConfirmedCases  
5  29-04-2020      Fatalities  
6  30-04-2020  ConfirmedCases  
7  30-04-2020      Fatalities  
8  01-05-2020  ConfirmedCases  
9  01-05-2020      Fatalities

In [5]:
# to get the summary of the dataframe
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 969640 entries, 0 to 969639
Data columns (total 9 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Id              969640 non-null  int64  
 1   County          880040 non-null  object 
 2   Province_State  917280 non-null  object 
 3   Country_Region  969640 non-null  object 
 4   Population      969640 non-null  int64  
 5   Weight          969640 non-null  float64
 6   Date            969640 non-null  object 
 7   Target          969640 non-null  object 
 8   TargetValue     969640 non-null  int64  
dtypes: float64(1), int64(3), object(5)
memory usage: 66.6+ MB


here, we can clearly see our data contains total 9 columns and total 969640 rows with some missing values in column country and province_state

In [6]:
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 311670 entries, 0 to 311669
Data columns (total 8 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ForecastId      311670 non-null  int64  
 1   County          282870 non-null  object 
 2   Province_State  294840 non-null  object 
 3   Country_Region  311670 non-null  object 
 4   Population      311670 non-null  int64  
 5   Weight          311670 non-null  float64
 6   Date            311670 non-null  object 
 7   Target          311670 non-null  object 
dtypes: float64(1), int64(2), object(5)
memory usage: 19.0+ MB


as we can see in the above shell our data contains total 8 columns and total 311670 rows with some missing values in column country and province_state

In [7]:
for i in test_df.columns:
    print(f"Unique{i}'s count:{test_df[i].nunique()}")
    print(f"{test_df[i].unique()}\n")

UniqueForecastId's count:311670
[     1      2      3 ... 311668 311669 311670]

UniqueCounty's count:1840
[nan 'Autauga' 'Baldwin' ... 'Uinta' 'Washakie' 'Weston']

UniqueProvince_State's count:133
[nan 'Australian Capital Territory' 'New South Wales' 'Northern Territory'
 'Queensland' 'South Australia' 'Tasmania' 'Victoria' 'Western Australia'
 'Alberta' 'British Columbia' 'Manitoba' 'New Brunswick'
 'Newfoundland and Labrador' 'Northwest Territories' 'Nova Scotia'
 'Ontario' 'Prince Edward Island' 'Quebec' 'Saskatchewan' 'Yukon' 'Anhui'
 'Beijing' 'Chongqing' 'Fujian' 'Gansu' 'Guangdong' 'Guangxi' 'Guizhou'
 'Hainan' 'Hebei' 'Heilongjiang' 'Henan' 'Hong Kong' 'Hubei' 'Hunan'
 'Inner Mongolia' 'Jiangsu' 'Jiangxi' 'Jilin' 'Liaoning' 'Macau' 'Ningxia'
 'Qinghai' 'Shaanxi' 'Shandong' 'Shanghai' 'Shanxi' 'Sichuan' 'Tianjin'
 'Tibet' 'Xinjiang' 'Yunnan' 'Zhejiang' 'Faroe Islands' 'Greenland'
 'French Guiana' 'French Polynesia' 'Guadeloupe' 'Martinique' 'Mayotte'
 'New Caledonia' 'Reunion'

In [8]:
for i in train_df.columns:
    print(f"Unique{i}'s count:{train_df[i].nunique()}")
    print(f"{train_df[i].unique()}\n")

UniqueId's count:969640
[     1      2      3 ... 969638 969639 969640]

UniqueCounty's count:1840
[nan 'Autauga' 'Baldwin' ... 'Uinta' 'Washakie' 'Weston']

UniqueProvince_State's count:133
[nan 'Australian Capital Territory' 'New South Wales' 'Northern Territory'
 'Queensland' 'South Australia' 'Tasmania' 'Victoria' 'Western Australia'
 'Alberta' 'British Columbia' 'Manitoba' 'New Brunswick'
 'Newfoundland and Labrador' 'Northwest Territories' 'Nova Scotia'
 'Ontario' 'Prince Edward Island' 'Quebec' 'Saskatchewan' 'Yukon' 'Anhui'
 'Beijing' 'Chongqing' 'Fujian' 'Gansu' 'Guangdong' 'Guangxi' 'Guizhou'
 'Hainan' 'Hebei' 'Heilongjiang' 'Henan' 'Hong Kong' 'Hubei' 'Hunan'
 'Inner Mongolia' 'Jiangsu' 'Jiangxi' 'Jilin' 'Liaoning' 'Macau' 'Ningxia'
 'Qinghai' 'Shaanxi' 'Shandong' 'Shanghai' 'Shanxi' 'Sichuan' 'Tianjin'
 'Tibet' 'Xinjiang' 'Yunnan' 'Zhejiang' 'Faroe Islands' 'Greenland'
 'French Guiana' 'French Polynesia' 'Guadeloupe' 'Martinique' 'Mayotte'
 'New Caledonia' 'Reunion' 'Saint 

In [ ]:
train_df[]